src/extension.ts

```diff
@@ -592,3 +597,6 @@ export function _isValidFileForAnalysis(documentUri: vscode.Uri) {
 	return allowedFileTypes.includes(path.extname(documentUri.path));
 }
 
+export function extractDiagnosticCode(diagnostic: vscode.Diagnostic) {
+	return typeof diagnostic.code === 'object' && 'value' in diagnostic.code ? String(diagnostic.code.value) : '';
```

Comment:
IIRC, using the `String` constructor is generally discouraged. Is there a `.toString()` method you can call, or something similar?